catalog/to-consul/syncer_test.go

```diff
@@ -155,6 +155,45 @@ func TestConsulSyncer_reapService(t *testing.T) {
 	}
 }
 
+// Test that the syncer doesn't reap any services until the initial sync has
+// been performed.
+func TestConsulSyncer_noReapingUntilInitialSync(t *testing.T) {
+	t.Parallel()
+
+	a, err := testutil.NewTestServerT(t)
+	require.NoError(t, err)
+	defer a.Stop()
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPAddr,
+	})
+	require.NoError(t, err)
+	s, closer := testConsulSyncer(client)
+	defer closer()
+
+	// Create a service directly in Consul. Since it was created directly we
+	// expect it to be deleted but not until the initial sync is performed.
+	svc := testRegistration(ConsulSyncNodeName, "baz", "default")
+	_, err = client.Catalog().Register(svc, nil)
+	require.NoError(t, err)
+
+	// We wait until the syncer has had the time to delete the service.
+	time.Sleep(200 * time.Millisecond)
```

Comment:
It seems like we might run into edge cases having this value exactly equal the `SyncPeriod` in the test ConsulSyncer. What about setting up the ConsulSyncer with an even smaller interval?

Reply:
Yes! Great idea. I didn't like this sleep but a retry loop doesn't work because it might run through before syncFull has run :D